tests/db/migration_tests/test_v202505090_to_v202505150.py

```diff
@@ -0,0 +1,44 @@
+"""
+Copyright 2025 Inmanta
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Contact: code@inmanta.com
+"""
+
+import os
+import re
+from collections import abc
+
+import asyncpg
+import pytest
+
+from inmanta import const, data
+
+file_name_regex = re.compile("test_v([0-9]{9})_to_v[0-9]{9}")
+part = file_name_regex.match(__name__)[1]
+
+
+@pytest.mark.db_restore_dump(os.path.join(os.path.dirname(__file__), f"dumps/v{part}.sql"))
+async def test_add_is_undefined_to_resource_table(
+    postgresql_client: asyncpg.Connection,
+    migrate_db_from: abc.Callable[[], abc.Awaitable[None]],
+) -> None:
+    # This migration script adds the is_undefined column to the resource table
+    # It should be set to true iff the status of the resource is undefined
+
+    await migrate_db_from()
+
+    resources = await data.Resource.get_list()
+    for resource in resources:
+        assert resource.is_undefined == (resource.status == const.ResourceState.undefined)
```

Comment:
this will break if we remove the status field, I think it is more interesting to pull two resource via the api, one which should be undefined and one which should not be undefined and assert they are as they should be.